overcast/runner/__init__.py

```diff
@@ -413,16 +457,35 @@ def provision_step(self, details):
         for base_node_name, node_info in stack['nodes'].items():
             def _create_node(base_name):
                 node_name = self.add_prefix(base_name)
-                self.nodes[base_name] = self.create_node(node_name, node_info,
-                                                         keypair=keypair_name,
-                                                         userdata=userdata)
+                self.nodes[base_name] = node(node_name, node_info,
+                                                        keypair=keypair_name,
+                                                        userdata=userdata,
+                                                        mappings=load_mappings(args.mappings))
+                self.nodes[base_name].build()
 
             if 'number' in node_info:
                 count = node_info.pop('number')
                 for idx in range(1, count+1):
-                    _create_node('%s%d' % (base_node_name, idx))
+                    node_name = '%s%d' % (base_node_name, idx)
+                    _create_node(node_name)
+                    node_names.add(node_name)
             else:
                 _create_node(base_node_name)
+                node_names.add(base_node_name)
+
+        while node_names:
+            time.sleep(5)
+            for name in node_names:
+                if self.nodes[name].poll() == 'ACTIVE':
+                    node_names.discard(name)
+                if self.nodes[name].poll() == 'ERROR':
+                    if details.get('retry-if-fails', False):
```

Comment:
I dont think so, it should be a setting for entire privisoning, so not only vms, but for others also, but im only handling it for vm at this moment. 

What i thought that I should be able to say that hey, I need you to retry if any step in the provisioning fail, also there is a timeout before which you should complete the provisioning. Since We are not using "number of retries" but using "timeout" I dont think this need to be per node or per object.